crates/worker/src/partition_processor_manager.rs

```diff
@@ -989,6 +989,7 @@ impl PartitionProcessorManager {
                     snapshot_base_path,
                     partition_store_manager: self.partition_store_manager.clone(),
                     cluster_name: config.common.cluster_name().into(),
+                    cluster_id: Metadata::current().nodes_config_ref().cluster_id(),
```

Comment:
I opted to just grab it from here, but if this proves popular we can provide a top-level convenience method similar to `my_node_id()`.